src/ast/mod.rs

```diff
@@ -753,8 +865,33 @@ impl FunctionDecl {
     }
 }
 
-impl<Head: Display + ToSexp, Leaf: Display + ToSexp, Ann> ToSexp
-    for GenericFunctionDecl<Head, Leaf, Ann>
+impl<Head, Leaf, Ann> GenericFunctionDecl<Head, Leaf, Ann>
+where
+    Ann: Clone + Default,
+    Head: Clone + Display,
+    Leaf: Clone + PartialEq + Eq + Display + Hash,
+{
+    pub fn map_exprs(
+        self,
+        f: &mut impl FnMut(GenericExpr<Head, Leaf, Ann>) -> GenericExpr<Head, Leaf, Ann>,
+    ) -> GenericFunctionDecl<Head, Leaf, Ann> {
+        GenericFunctionDecl {
```

Comment:
Using [this syntax](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax) might make this (and a few other places) look cleaner

Reply:
True, but then you might forget to update this code if things change in the future
Keeping as is for now